codespell_lib/tests/test_basic.py

```diff
@@ -455,6 +455,86 @@ def test_context(tmpdir, capsys):
     assert 'ERROR' in lines[0]
 
 
+def test_ignore_regex_flag(tmpdir, capsys):
+    """Test ignore regex flag functionality."""
+    d = str(tmpdir)
+
+    # Invalid regex.
+    code, stdout, _ = cs.main('--ignore-regex=(', std=True)
+    assert code == EX_USAGE
+    assert 'usage:' in stdout
+
+    # Empty regex matches everything.
+    with open(op.join(d, 'flag.txt'), 'w') as f:
+        f.write('# Please see http://example.com/abandonned for info\n')
+    assert cs.main(f.name, '--ignore-regex=^$') == 1
+
+    # Custom ignore.
+    with open(op.join(d, 'flag.txt'), 'w') as f:
+        f.write('abandonned\n')
+    assert cs.main(f.name, '--ignore-regex=abandonned') == 0
+
+
+def test_uri(tmpdir, capsys):
+    """Test ignore regex functionality for URIs."""
+    d = str(tmpdir)
+
+    # Ignoring text in path.
+    with open(op.join(d, 'uri.txt'), 'w') as f:
+        f.write('# Please see http://example.com/abandonned for info\n')
+    assert cs.main(f.name) == 0
+
+    # Test a different protocol.
+    with open(op.join(d, 'uri.txt'), 'w') as f:
+        f.write('# Please see https://example.com/abandonned for info\n')
+    assert cs.main(f.name) == 0
+
+    # Ignoring text in path ending with /.
+    with open(op.join(d, 'uri.txt'), 'w') as f:
+        f.write('# Please see http://example.com/abandonned/ for info\n')
+    assert cs.main(f.name) == 0
+
+    # Ignoring text in domain.
+    with open(op.join(d, 'uri.txt'), 'w') as f:
+        f.write('# Please see http://abandonned.com/example for info\n')
+    assert cs.main(f.name) == 0
+
+    # Ignoring text in anchor.
+    with open(op.join(d, 'uri.txt'), 'w') as f:
+        f.write('# Please see http://example.com/ex#abandonned for info\n')
+    assert cs.main(f.name) == 0
+
+    # Typo because there's no protocol.
+    with open(op.join(d, 'uri.txt'), 'w') as f:
+        f.write('# Please see example.com/abandonned for info\n')
+    assert cs.main(f.name) == 1
+
+    # Typo because there aren't enough domain parts.
+    with open(op.join(d, 'uri.txt'), 'w') as f:
+        f.write('# Please see http://abandonned for info\n')
```

Comment:
This would be fine in some scenarios, for example within a company.

Reply:
True, although it didn't feel like it'd be a common enough issue to handle. Let me know if you'd rather support this style (it was deliberate as an example). But removed here.

Reply:
I think a lot of large businesses will have stuff that's not FQDN, also things like open source projects (although this not great example uses pi.hole):
https://docs.pi-hole.net/guides/vpn/only-dns-via-vpn/#alternative-2-set-dns-server-address-of-your-pi-hole-manually-on-the-main-interface

Reply:
Resolving from this PR anyway.